assembly/src/assembler/context.rs

```diff
@@ -88,6 +88,16 @@ impl AssemblyContext {
         }
     }
 
+    /// Returns the [Procedure] by its index from the vector of compiled procedures.
+    pub fn get_compiled_procedure(&self, idx: u16) -> Result<&Procedure, AssemblyError> {
```

Comment:
nit: I would probably rename this to `get_local_procedure`.

Also, curious: in the procedure right above this one (`get_imported_procedure_name`) we say `self.module_stack.first()` and here we say ` self.module_stack.last()`. Seems like `.last()` is the right thing to do - so, is there a bug in `get_imported_procedure_name()`?

Reply:
I also noticed that. Not sure for now, I should dig into it once again and test it to be sure, but preliminarily I think that `.last()` is the correct approach, since that's how we get the stack top if it is implemented on vector.

Reply:
It was indeed a bug. I added another test (`get_proc_name_of_unknown_module()`) to check this.